packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-inline-checkout-button-strategy.ts

```diff
@@ -196,7 +196,13 @@ export default class PaypalCommerceInlineCheckoutButtonStrategy implements Check
         methodId: string,
         callback?: () => void
     ): Promise<void> {
-        await this._submitPayment(methodId, data.orderID);
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const { intent } = paymentMethod.initializationData;
+
+        if (intent === 'capture') {
```

Comment:
Are there many `intent` types? I am asking since if we know that there can be a few then maybe we can create an intent Enum?

Reply:
We have 2 possible values for intent: 'capture' and 'authorize'. I think we can make an Enum for that

Reply:
Done 👍